src/HealthChecks.AzureStorage/AzureFileShareHealthCheck.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Azure.Storage.Files.Shares;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace HealthChecks.AzureStorage
+{
+    public class AzureFileShareHealthCheck : IHealthCheck
+    {
+        private readonly string _connectionString;
+        private readonly string _shareName;
+
+        public AzureFileShareHealthCheck(string connectionString, string shareName = default)
+        {
+            _connectionString = connectionString ?? throw new ArgumentNullException(nameof(connectionString));
+            _shareName = shareName;
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                var shareClient = new ShareClient(_connectionString, _shareName);
```

Comment:
Are you sure you want to create an instance of `ShareClient` each healthcheck request?

Reply:
Fixed it